src/shared/lib/timers/TimerBase.ts

```diff
@@ -25,12 +23,14 @@ abstract class TimerBase {
 
   abstract setTimer(duration?: number): void;
 
-  protected onForeground?(): void;
+  protected abstract onForeground(): void;
   protected onBackground?(): void;
 
-  public stop(): void {
+  public stop(isRestart = false): void {
     this.hasStarted = false;
-    this.removeAppStateListener();
+    if (!isRestart) {
+      this.removeAppStateListener();
+    }
   }
```

Comment:
Instead of using `isRestrart` flag, we can add a new listener in `start` method instead. 
I mean, we can move the creation of a listener from `constructor` to `start`.